<launch>
  <arg name="camera"      default="camera"/>
  <arg name="draw_markers" default="false" />
  <arg name="map_file" default="exp_test.json" />

  <include file="$(find turtlebot_bringup)/launch/3dsensor.launch"/>
  <include file="$(find turtlebot_bringup)/launch/minimal.launch"/>

  <node pkg="nodelet" type="nodelet" name="tracker_nodelet" args="load mediax_project/TrackerNodelet $(arg camera)/$(arg camera)_nodelet_manager" output="screen">
	  <param name="camera_info_file" value="$(find mediax_project)/camera_calibration/kinect_rgb.yml"/>
	  <param name="map_file" value="$(find mediax_project)/maps/marker_maps/$(arg map_file)"/>
	  <param name="filter_param_file" value="$(find mediax_project)/params/robot_tracker.json"/>
	  <param name="marker_size" value="0.204"/>
	  <param name="filter_mode" value="kalman"/>
      <param name="odom_source" value="odometer"/>
	  <param name="num_frames_skip" value="2"/>
	  <param name="num_process_skip" value="1"/>
	  <param name="image_scale" value="1.0"/>
	  <param name="draw_markers" value="$(arg draw_markers)"/>
	  <remap from="/$(arg camera)/inside_out_tracker/image" to="/$(arg camera)/rgb/image_mono"/>
	  <remap from="/$(arg camera)/inside_out_tracker/pose2d" to="/tracking/robot_pose2d"/>
	  <remap from="/$(arg camera)/inside_out_tracker/odom" to="/odom"/>
	  <remap from="/$(arg camera)/inside_out_tracker/reset" to="/tracking/reset_robot_tracker"/>
  </node>

  <!--node pkg="mediax_project" type="imu_manager_node" name="imu_manager" output="screen">
	<remap from="/imu_data_raw" to="/tracking/imu_data_raw"/>
  </node-->
</launch>
